notebooks/segmentation/segmentation_inspection.ipynb

```diff
@@ -362,6 +392,403 @@
+    "        \n",
+    "        Returns\n",
+    "        -------\n",
+    "        video_utils.ThumbnailVideo\n",
+    "        \"\"\"\n",
+    "        if origin is None:\n",
+    "            origin = (0, 0)\n",
+    "        if frame_shape is None:\n",
+    "            frame_shape = self.video_shape[1:3]\n",
+    "\n",
+    "        thumbnail = video_utils.thumbnail_video_from_path(\n",
+    "                        self.video_path,\n",
+    "                        origin,\n",
+    "                        frame_shape,\n",
+    "                        timesteps=timesteps,\n",
+    "                        tmp_dir=self.tmp_dir,\n",
+    "                        fps=fps,\n",
+    "                        quality=quality,\n",
+    "                        normalization=self.vmax)\n",
+    "        return thumbnail\n",
+    "\n",
+    "    def get_thumbnail_video_from_roi(self,\n",
+    "                                     roi=None,\n",
+    "                                     roi_color=None,\n",
+    "                                     timesteps=None,\n",
+    "                                     quality=5,\n",
+    "                                     fps=31):\n",
+    "        \"\"\"\n",
+    "        Get a ThumbnailVideo from an ROI\n",
+    "        \n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        roi: ExtractROI\n",
+    "        \n",
+    "        roi_color: Optional[Tuple[int, int, int]]\n",
+    "            If not None, the RGB color in which to plot the ROI's\n",
+    "            boundary. If None, ROI is not plotted in thumbnail.\n",
+    "            (default: None)\n",
+    "        \n",
+    "        timesteps: Optional[np.ndarray]\n",
+    "            If not None, timesteps to put in the thumbnail\n",
+    "            video. If None, use all timesetps (default: None)\n",
+    "        \n",
+    "        fps: int\n",
+    "            frames per second (default: 31)\n",
+    "        \n",
+    "        quality: int\n",
+    "            quality parameter passed to imageio. Max is 10.\n",
+    "            (default: 5)\n",
+    "        \n",
+    "        Returns\n",
+    "        -------\n",
+    "        video_utils.ThumbnailVideo\n",
+    "        \"\"\"\n",
+    "        thumbnail = video_utils.thumbnail_video_from_ROI(\n",
+    "                        self.video_path,\n",
+    "                        roi,\n",
+    "                        roi_color=roi_color,\n",
+    "                        timesteps=timesteps,\n",
+    "                        tmp_dir=self.tmp_dir,\n",
+    "                        fps=fps,\n",
+    "                        quality=quality,\n",
+    "                        normalization=self.vmax)\n",
+    "        return thumbnail"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import Video\n",
+    "\n",
+    "class VideoDisplayGenerator(object):\n",
```

Comment:
similarly, this class cuold take `tmpdir` as an arg.
I think you have it in the notebook so that `IPython.display.Video` doesn't need to be an ophys_etl requirement. If you changed the `display_video` method to return the tuple `(plot_path, width=width, height=height, embed=False)` then the notebook could just do:
`Video(*Generator.display())`
and we could move more code out of the notebook.